cpp/examples/arrow/execution_plan_documentation_examples.cc

```diff
@@ -761,6 +759,30 @@ arrow::Status TableSinkExample() {
   std::cout << "Results : " << output_table->ToString() << std::endl;
   return arrow::Status::OK();
 }
+
+// (Doc section: Table Sink Example)
+
```

Comment:
```suggestion
```

Reply:
Ah, shouldn't we put this in either side of the code-segment that we want to refer to?

For instance

```c++
// (Doc section: Table Sink Example)

/// Code refer to

// (Doc section: Table Sink Example)
```
Even if this is the correct way to do, there is a typo in my code in the `RecordBatchReaderSource` section.

`// (Doc section: RecordBatchReaderSource Example)`, I haven't closed that properly. I did close it now.

Reply:
Yes, sorry.  I hadn't realized this was missing before.  You are correct about the Table Sink Example part.